packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -117,6 +117,40 @@ fn process_fn_items(
         }])
         .collect::<Vec<proc_macro2::TokenStream>>();
 
+    // Take a second pass over `abi_types` in order to update an `TypeDeclarations` that need to
+    // be updated with more information for the codegen process.
+    let abi_types = abi_types
+        .iter()
+        .map(|typ| {
+            // If this is an array type we have to manually update it's type field to include its inner
+            // type so that the rest of the codegen will work with minimal changes.
+            if is_array_type(typ) {
+                let inner = typ.components.as_ref().unwrap().first().unwrap().type_id;
+                let size = typ
+                    .type_field
+                    .split(' ')
+                    .last()
+                    .unwrap()
+                    .trim_end_matches(']')
+                    .parse::<usize>()
+                    .unwrap();
```

Comment:
Can we switch these unwraps for expects? Seems that the other additions have expects, so we should be consistent.